manifests/telegraf.pp

```diff
@@ -85,6 +88,7 @@
   String  $uid,
   String  $user,
   String  $user_comment,
+  Optional[String] $gid = undef,
```

Comment:
I think it's good to avoid using Optional variables set to undef since then the variable gets hidden if people just search `data/common.yaml`.

Since we are already using `$gid` as a string, what if we set something like this in `data/common.yaml` `profile_monitoring::telegraf::gid: "first_available"`. Then change the `if ( $gid )` condition to check if `gid` matches that string instead.

Reply:
I think it is better to continue using `null`/`undef`, but specifically set it to `null` by default in the hiera data.

Reply:
True that's even better